lib/table.js

```diff
@@ -271,10 +271,15 @@ internals.validateItemFragment = (item, schema) => {
   );
 
   if (updateValidation.error) {
-    const errors = _.omitBy(
-      updateValidation.error.details,
-      e => _.isEqual(e.type, 'any.required')
-    );
+    let errors = [];
+    if (!updateValidation.error.details) {
```

Comment:
Please add comment on what exactly this conditional is checking. i.e. "If someone overrides the Joi errors with a custom Error object we won't have a `details` property..." or something along those lines.

I would also add it to the `create()` method if it is using the same logic.